### PR TITLE
Fix for python 3.8 CI.

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

We are seeing the python 3.8 CI failing, e.g. https://github.com/astropy/astropy/actions/runs/3084051677/jobs/4985734475. This appears to be a result of the `pandas` 1.5 release, which requires `numpy` 1.20.3 or later. This messes up the `oldest-supported-numpy` requirement causing the latest `numpy` to be installed.

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/timeseries/tests/test_sampled.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def test_fold_invalid_options():
 def test_pandas():
     pandas = pytest.importorskip("pandas")
 
-    PANDAS_LT_1_5 = not minversion(pandas, '1.4.999')  # When this was written 1.5 was not released.
+    PANDAS_LT_1_5 = minversion(pandas, '1.5')  # When this was written 1.5 was not released.
```
Comment (mhvk) [inline astropy/timeseries/tests/test_sampled.py:275]: Remove the comment, I guess!

**Review by mhvk — COMMENTED** [review]
Thanks for fixing this! Just a nitpick.

File: astropy/timeseries/tests/test_sampled.py (hunk 1 above)
Comment (pllim) [inline astropy/timeseries/tests/test_sampled.py:275]: Don't take out `not`
```suggestion
    PANDAS_LT_1_5 = not minversion(pandas, '1.5')
```

File: docs/timeseries/pandas.rst [hunk 2]
```diff
@@ -33,7 +33,7 @@ Consider a concise example starting from a :class:`~pandas.DataFrame`:
     >>> import pandas
     >>> import numpy as np
     >>> from astropy.utils.introspection import minversion
-    >>> PANDAS_LT_1_5 = not minversion(pandas, '1.4.999')
+    >>> PANDAS_LT_1_5 = minversion(pandas, '1.5')
```
Comment (pllim) [inline docs/timeseries/pandas.rst:36]:
```suggestion
    >>> PANDAS_LT_1_5 = not minversion(pandas, '1.5')
```

**Comment by pllim** [comment]
Would need a squash before merge. 😉

**Comment by WilliamJamieson** [comment]
Doing some digging shows that pandas-dev/pandas#48417 reverted the changes which made #13573 necessary.

File: astropy/timeseries/tests/test_sampled.py (hunk 1 above)
Comment (WilliamJamieson) [reply astropy/timeseries/tests/test_sampled.py]: Turns out this check isn't necessary.

**Review by pllim — APPROVED** [review]
Thanks for the quick fix! 🙇‍♀️

**Pull request merged**